File: Proyecto_1/Proyecto_1/src/analizadores/nodo.java
package analizadores;
import java.util.ArrayList;

public class nodo {
    
    private String etiqueta;
    private ArrayList<nodo> hijos = new ArrayList<>();
    public String valor = null;
    public boolean cadena = false;
    private int id ;

    public nodo (String etiqueta, int id){
        this.etiqueta = etiqueta;
        this.id = id;
    
    }
    
    public nodo (String etiqueta, String valor, int id){
        this.etiqueta = etiqueta;
        this.id = id;
        this.valor = valor;
    
    }
    
    public void AddHijos(nodo nuevo) {
        hijos.add(nuevo);
    }
    
    public String getEtiqueta() {
        return etiqueta;
    }

    public void setEtiqueta(String etiqueta) {
        this.etiqueta = etiqueta;
    }

    public ArrayList<nodo> getHijos() {
        return hijos;
    }
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id; 
    }
    
    public String getValor() {
        return valor;
    }
    
    public void setValor(String valor) {
        this.valor = valor; 
    }
    
    
}

File: Proyecto_2/nodo.ts
//creamos una clase
export class Nodo{

    //atributos
    private etiqueta: string=""
    private hijos: Array<Nodo>
    private valor: string=""
    private cadena: boolean = false
    private aux: string=""
    private id: number


    // constructor 1
    private constructor(         
        private etiqueta_: string,
        private id_: number,
        private valor_: string


    ){ 
        this.etiqueta = etiqueta_
        this.valor = valor_
        this.id = id_
    }

    public addHijos(nuevo: Nodo ):void{
       let hijosv =  this.hijos.push(nuevo)
       console.log(this.hijos)
       console.log(hijosv)
    }

    public setEtiqueta(eti: string):void{
        this.etiqueta = eti      
    }

    public setValor(val: string):void{
        this.valor = val      
    }

    public setId(identificador: number):void{
        this.id = identificador     
    }


    public getHijos():Array<Nodo>{
        return this.hijos
    }


    public getEtiqueta():string{
        return this.etiqueta
    }

    public getValor():string{
        return this.valor
    }

    public getId():number{
        return this.id
    }

}